routers/repo/view.go

```diff
@@ -307,6 +307,12 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 			var output bytes.Buffer
 			lines := strings.Split(fileContent, "\n")
 			ctx.Data["NumLines"] = len(lines)
+			ctx.Data["NumLinesZero"] = false
+			if len(lines) == 1 && lines[0] == "" {
+				// If the file is completely empty, we show zero lines at the line counter
+				ctx.Data["NumLines"] = 0
+				ctx.Data["NumLinesZero"] = true
```

Comment:
This looks a little hacky. I'd consolidate the information into something like:
```
ctx.Data["LineCountInfoAvailable"] = true
```

and then the template will look like:

```
{{if .LineCountInfoAvailable}}
```
The effect is the same, only less hacky.

The name `LineCountInfoAvailable` is only to convey the meaning in this comment, you'll probably come up with a better name. 😁

Reply:
I agree, this looks a lot cleaner than my solution. 👍